packages/ragbits-document-search/src/ragbits/document_search/_main.py

```diff
@@ -141,8 +141,26 @@ async def ingest(
         elements = await self.processing_strategy.process_documents(
             documents, self.document_processor_router, document_processor
         )
+        await self.remove_entries_with_same_sources(elements)
         await self.insert_elements(elements)
 
+    async def remove_entries_with_same_sources(self, elements: list[Element]) -> None:
+        """
+        Remove entries from the vector store whose source id is present in the elements' metadata.
+
+        Args:
+            elements: List of elements whose source ids will be checked and removed from the vector store if present.
+        """
+        unique_source_ids = {element.document_meta.source.id for element in elements}
+
+        ids_to_delete = []
+        for entry in await self.vector_store.list():
```

Comment:
Shouldn't we use a where clause here to limit which elements we want to retrieve from the vector store? Fetching everything is not optimal.

Also, I noticed that in the Qdrant implementation, the list by default returns only 10 rows. We should probably change that to make this feature functional.

Reply:
Regarding Qdrant limit: `AsyncQdrantClient.query_points()` takes `limit` argument which needs to be of `int` type. Default value of `limit` is set to 10 there, so it makes sense to pass `limit=limit or 10` when calling `query_points`, because limit is still set to 10 if not provided and it passes mypy checks. Maybe it's possible to ignore mypy there, but I haven't tested the functionality when `limit=None` is passed.

Reply:
Regarding listing all entries: we need to check what is the source of every entry in the vector store. I can't think of a better way to do this than by retrieving all the entries

Reply:
Added checking if `limit` is `None` and if so, then setting `limit` to the number of the entries in the vector store.
Added `TODO` for passing `where` argument